pkg/service/tenant.go

```diff
@@ -44,7 +50,24 @@ func (s *tenantService) Update(ctx context.Context, req *v1.TenantUpdateRequest)
 func (s *tenantService) Delete(ctx context.Context, req *v1.TenantDeleteRequest) (*v1.TenantResponse, error) {
 	tenant := req.NewTenant()
 	err := s.tenantStore.Delete(ctx, tenant.Meta.Id)
-	return tenant.NewTenantResponse(), err
+	if err != nil {
+		return nil, err
+	}
+	filter := map[string]any{
+		"tenantmember ->> 'tenant_id'": tenant.Meta.Id,
+		"tenantmember ->> 'member_id'": tenant.Meta.Id,
```

Comment:
this is an and, unsure but isnt it suffice to just search for member_id == tenant.Meta.Id ?

Reply:
I wasn't sure about this. From the way the filter was used (here)[https://github.com/metal-stack/masterdata-api/blob/master/pkg/service/tenantmember.go#L80] I assumed it was an OR. If it is an AND, I guess there is no better way than using two separate queries, right?

Reply:
done